symengine/basic.h

```diff
@@ -97,7 +97,15 @@ class Basic : public EnableRCPFromThis<Basic>
     mutable hash_t hash_; // This holds the hash value
 #endif // WITH_SYMENGINE_THREAD_SAFE
 public:
+#ifdef USE_VIRTUAL_TYPEID
     virtual TypeID get_type_code() const = 0;
+#else
+    TypeID type_code_;
+    inline TypeID get_type_code() const
+    {
+        return type_code_;
```

Comment:
It wouldn't work because then `type_code_` would be declared in the child class and would require a virtual function, but that is what we are trying to avoid.

Initializer list does not make a difference for types with empty constructors like `enum`s